src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStoreV1.cs

```diff
@@ -276,5 +276,64 @@ private async Task DeleteInstanceAsync(string studyInstanceUid, string seriesIns
                 }
             }
         }
+
+        public async Task<int> RetrieveNumDeletedExceedRetryCountAsync(int maxRetryCount, CancellationToken cancellationToken = default)
+        {
+            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                sqlCommandWrapper.CommandText = @$"
+                        SELECT COUNT(Case When {VLatest.DeletedInstance.RetryCount} >= {maxRetryCount} THEN 1 ELSE NULL END)
+                        FROM {VLatest.DeletedInstance.TableName}";
+
+                try
+                {
+                    using (SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(cancellationToken))
+                    {
+                        while (await sqlDataReader.ReadAsync(cancellationToken))
```

Comment:
the query only return 1 row data, I'm thinking if we should use `if` instead of `while`